account_fiscal_position_partner_type/models/account_fiscal_position.py

```diff
@@ -11,24 +11,20 @@ class AccountFiscalPosition(models.Model):
     fiscal_position_type = fields.Selection(
         selection=[("b2c", "End customer (B2C)"), ("b2b", "Company (B2B)")],
         string="Type",
-        default=lambda self: self._default_fiscal_position_type(),
+        default=lambda self: self.env.company.default_fiscal_position_type,
     )
 
-    @api.model
-    def _default_fiscal_position_type(self):
-        return self.env.company.default_fiscal_position_type
-
     @api.model
     def search(self, args, offset=0, limit=None, order=None, count=False):
-        if self.env.context.get("fiscal_position_type"):
+        if "fiscal_position_type" in self.env.context:
```

Comment:
This changes behavior if contact has {"fiscal_position_type": False}
Don't change it please.

Reply:
Done the suggested change.
Thank you.